### PR TITLE
hark: follow individual indices 

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Follow individual indices in graph-store, and notify when an immediate child is created. In order to bypass potentially empty indices the notification-kind type is now `(unit [@t @ud])` where the `@ud` describes the length of the parent index.

### CODE REVIEW

**Comment by matildepark** [comment]
I'm confused how there are even conflicts on this one.

File: pkg/arvo/app/hark-graph-hook.hoon [hunk 1]
```diff
@@ -83,16 +83,16 @@
       %set-watch-on-self  (set-watch-on-self +.action)
     ==
     ++  listen
-      |=  graph=resource
+      |=  [graph=resource =index:post]
       ^-  (quip card _state)
-      :-  (give:ha ~[/updates] [%listen graph])
-      state(watching (~(put in watching) graph))
+      :-  (give:ha ~[/updates] [%listen graph index])
```
Comment (tacryt-socryp) [inline pkg/arvo/app/hark-graph-hook.hoon:88]: Same comment about give being factored out, of course.

File: pkg/arvo/app/hark-graph-hook.hoon [hunk 2]
```diff
@@ -187,15 +195,19 @@
...
       ?~  notif-kind
         [child-cards state]
       =/  desc=@t
         ?:  (is-mention contents.post.node)
           %mention 
-        u.notif-kind
+        name.u.notif-kind
+      =/  parent=index:post
+        (scag parent-lent.u.notif-kind index.post.node)
+      ~&  parent
```
Comment (tacryt-socryp) [inline pkg/arvo/app/hark-graph-hook.hoon:208]: Remove

File: pkg/arvo/lib/graph-store.hoon [hunk 3]
```diff
@@ -52,6 +52,7 @@
   ++  index
     |=  i=^index
     ^-  json
+    ?:  =(~ i)  s+'/'
```
Comment (tacryt-socryp) [inline pkg/arvo/lib/graph-store.hoon:55]: good

**Review by tacryt-socryp — APPROVED** [review]
LGTM

**Comment by matildepark** [comment]
This was conflicting on, predominantly, `++  graph-index` vs `++  graph-indices` in lib/graph-hook from commits from 15 days ago, so I don't really get that. Will remove sigpam and merge shortly

**Comment by matildepark** [comment]
hark-graph-hook nest-fails, so will leave open — looks like on changes in the post-L review commit that weren't written with these in mind?

**Pull request merged**